src/pages/resultsView/pathwayMapper/ResultsViewPathwayMapper.tsx

```diff
@@ -0,0 +1,198 @@
+    });
+
+    addGenomicData: (alterationData: ICBioData[]) => void;
+
+    pathwayHandler: Function;
+
+    constructor(props: IResultsViewPathwayMapperProps){
+        super(props);
+        this.alterationFrequencyData = [];
+        this.isLoading = false;
+        this.activeToasts = [];
+        this.validGenesAccumulator = {};
+    }
+
+    render() {
+
+        // Alteration data of query genes are loaded
+        this.props.store.oqlFilteredCaseAggregatedDataByUnflattenedOQLLine.result!.forEach( (alterationData, trackIndex) => {
+
+            getAlterationData(this.props.store.samples.result,
+                              this.props.store.patients.result,
+                              this.props.store.coverageInformation.result,
+                              this.props.store.sequencedSampleKeysByGene.result!,
+                              this.props.store.sequencedPatientKeysByGene.result!,
+                              this.props.store.selectedMolecularProfiles.result!,
+                              alterationData,
+                              true,
+                              this.alterationFrequencyData,
+                              this.props.store.genes.result!);
+        });
+
+
+        // Alteration data of non-query genes are loaded.
+        const isNewStoreReady = this.storeForAllData &&
+                                this.storeForAllData.oqlFilteredCaseAggregatedDataByUnflattenedOQLLine.isComplete &&
+                                this.storeForAllData.samples.isComplete &&
+                                this.storeForAllData.patients.isComplete &&
+                                this.storeForAllData.coverageInformation.isComplete &&
+                                this.storeForAllData.sequencedSampleKeysByGene.isComplete &&
+                                this.storeForAllData.sequencedPatientKeysByGene.isComplete &&
+                                this.storeForAllData.selectedMolecularProfiles.isComplete;
+
+        if(isNewStoreReady){
+
+            this.storeForAllData!.oqlFilteredCaseAggregatedDataByUnflattenedOQLLine.result!.forEach( (alterationData, trackIndex) => {
+                getAlterationData(this.storeForAllData!.samples.result,
+                                  this.storeForAllData!.patients.result,
+                                  this.storeForAllData!.coverageInformation.result,
+                                  this.storeForAllData!.sequencedSampleKeysByGene.result!,
+                                  this.storeForAllData!.sequencedPatientKeysByGene.result!,
+                                  this.storeForAllData!.selectedMolecularProfiles.result!,
+                                  alterationData,
+                                  false,
+                                  this.alterationFrequencyData,
+                                  this.props.store.genes.result!);
+                
+            });
+
+            this.addGenomicData(this.alterationFrequencyData);
```

Comment:
I don't see where addGenomidData is actually implemented

Reply:
ok, i see, it gets set by addGenomicDataHandler.  The problem is though that the existence of addGenomicData method on this class is now part of state, and yet because it isn't observable, the component won't react to the resetting of that function.  It maybe be that it only gets set once, but nevertheless it seems like a matter of luck that it works. addGenomicData reference should be observable so that when it gets set by the child component, it will trigger a render cycle.

Reply:
Okay then, I will put observable decorator to `addGenomicData`.